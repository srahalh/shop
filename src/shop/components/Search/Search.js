import React from 'react';
import PropTypes from 'prop-types';

import './Search.scss';

const Search = ({setSearchValue}) => {
  return (
    <div className="Search__container">
      <input 
        type="text"
        label="search"
        placeholder="Search article by category or title"
        onChange={(event) => setSearchValue(event.target.value)}
      />
    </div>
  )
}

Search.propTypes = {
  setSearchValue: PropTypes.func
}

export default Search