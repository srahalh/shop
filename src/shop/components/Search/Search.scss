@import '../../styles/mixins.scss';
@import '../../styles/config.scss';

.Search__container {
  display: block;

  @include flex-layout-space-between;

  @include tablet {
    @include container-tablet;
    justify-content: center;
    margin-top: $margin-lg;
  }

  @include desktop {
    @include container-desktop;
    justify-content: end;
    margin-top: $margin-lg;
  }

  @include desktop-lg {
    @include container-desktop-lg;
    justify-content: end;
    margin-top: $margin-lg;
  }

  justify-content: center;
  margin-top: $margin-lg;

  input {
    border: 1px solid lighten($color-primary, 70%);
    height: 2rem;
    width: 300px;
    
    &:focus {
      outline: 0 none;
      border-color: $color-secondary;
    }

    @include desktop {
      flex: 0 0 40%;
    }

    @include desktop-lg {
      flex: 0 0 25%;
    }
  }
}