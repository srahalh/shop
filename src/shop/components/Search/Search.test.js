import React from "react";
import { render } from "@testing-library/react";
import Search from "./Search";

describe("Search component", () => {
  test("Should render Search Component", () => {
    const { asFragment  } = render(<Search />);
    expect(asFragment ()).toMatchSnapshot();
  });
});
