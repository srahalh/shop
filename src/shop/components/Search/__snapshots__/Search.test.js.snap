// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Should render Search Component 1`] = `
<DocumentFragment>
  <div
    class="Search__container"
  >
    <input
      label="search"
      placeholder="Search article by category or title"
      type="text"
    />
  </div>
</DocumentFragment>
`;
